nixos/modules/programs/dconf.nix

```
@@ -1,55 +1,174 @@
 { config, lib, pkgs, ... }:
 
-with lib;
-
 let
   cfg = config.programs.dconf;
-  cfgDir = pkgs.symlinkJoin {
-    name = "dconf-system-config";
-    paths = map (x: "${x}/etc/dconf") cfg.packages;
-    postBuild = ''
-      mkdir -p $out/profile
-      mkdir -p $out/db
-    '' + (
-      concatStringsSep "\n" (
-        mapAttrsToList (
-          name: path: ''
-            ln -s ${path} $out/profile/${name}
-          ''
-        ) cfg.profiles
-      )
-    ) + ''
-      ${pkgs.dconf}/bin/dconf update $out/db
-    '';
-  };
+
+  # Compile keyfiles to Dconf DB
+  compileDconfDb = dir: pkgs.runCommand "dconf-db"
+    {
+      nativeBuildInputs = [ pkgs.dconf ];
+    } "dconf compile $out ${dir}";
+
+  # Recursively lock values in settings
+  recuriseLock = with lib; settings: locks: (mapAttrsRecursive
+    (path: value: optionalAttrs value
+      (mapAttrsRecursiveCond (value: !lib.gvariant.isGVariant value)
+        (path: value: true)
+        (foldl'
+          (root: name: if hasAttr "${name}" root then root."${name}" else
+          throw "${concatStringsSep "." (tail path)} doesn't exist in settings.")
+          { v = settings; }
+          path)))
+    { v = locks; }).v;
```

It is very difficult to see which attributes belong to which function. Could you please indent it so it is clearer, for example:

```suggestion
  recuriseLock =
    settings:
    locks:

    (lib.mapAttrsRecursive
      (path:
      value:

      lib.optionalAttrs
        value
        (lib.mapAttrsRecursiveCond
          (value': !lib.gvariant.isGVariant value')
          (path': value': true)
          (foldl'
            (root: name: root."${name}" or throw "${lib.concatStringsSep "." (lib.tail path)} doesn't exist in settings.")
            { v = settings; }
            path
          )
        )
      )
      { v = locks; }).v;
```

Or consider splitting the function somehow.